### ISSUE TITLE
[BUG] inventree-dev-worker does not auto-reload on changed plugin files

### ISSUE THREAD

**Issue by simonkuehling** [inventree/InvenTree#2902]
**Describe the bug**
In the dev docker setup, the `inventree-dev-server` auto-reloads after files have changed in the codebase - which is neat and a huge timesaver for development. During testing a plugin in the `Inventree/plugins/` directory I noticed that while `inventree-dev-server` does reload upon each change as expected, `inventree-dev-worker` does not.
The plugin uses the EventMixin - so it is executed by the background worker and with that, never auto-reloads as far as I can tell.

**Steps to Reproduce**

1. Ad a plugin file to `./Inventree/plugins` folder
2. make a change to the file and save
3. observe `inventree-dev-server` reloading while `inventree-dev-worker` does not

**Expected behavior**
`inventree-dev-worker` should also auto-reload to speed up development, I guess?

**Deployment Method**
- [X] Docker
- [ ] Bare Metal

**Version Information**
# Version Information:
InvenTree-Version: 0.7.0 dev
Django Version: 3.2.13
Commit Hash: adfcd42e0
Commit Date: 2022-04-22
Database: postgresql
Debug-Mode: True
Deployed using Docker: True

**Comment by simonkuehling** [comment]
Oh, I see... I had hoped that it just wasn't configured...
Is it maybe possible to trigger a service restart of `inventree-dev-worker` from within `inventree-dev-server` somehow? But I'm just guessing here...

I'm happy to close this issue if it's not a quick fix right now.

**Comment by SchrodingersGat** [comment]
I don't think a quick fix is going to be possible for this one

**Comment by matmair** [comment]
@simonkuehling if you are trying to debug a plugin with events I would suggest to run your debug instance without workers. Then your tasks will run through your debugging instance and you can check what the tasks do exactly.

**Comment by SchrodingersGat** [comment]
@matmair good point. It might be worth adding a note about this to the plugin documentation.

**Comment by simonkuehling** [comment]
> @simonkuehling if you are trying to debug a plugin with events I would suggest to run your debug instance without workers. Then your tasks will run through your debugging instance and you can check what the tasks do exactly.

@matmair, do I just stop the worker service for that - or do I need to configure anything else in order for the server to take over those tasks?

**Comment by SchrodingersGat** [comment]
@simonkuehling just stop the worker service and any background tasks should be run by the web server (in a blocking fashion)

**Comment by simonkuehling** [comment]
@SchrodingersGat - just tried stopping the worker service with `docker-compose stop inventree-dev-worker` and restartet the server with `docker-compose restart inventree-dev-server` (just to be sure), but background tasks still seem to be queued and no plugin output is reported in the server logs so far. Am I doing it right?

**Comment by SchrodingersGat** [comment]
You might have to wait a few minutes before trying - the way that the server determines if the worker is running is to look for a heartbeat message which expires after ~20 minutes:

![image](https://user-images.githubusercontent.com/10080325/166933871-64c471f5-65a7-439f-991d-00a7ae51d09f.png)

So you could wait half an hour or delete the existing heartbeat tasks from the database.
